### PR TITLE
Api deprecation changes

### PR DESCRIPTION
**Pull request by nabhajit-ray**
### Description
[Api deprecation changes]

### Issues Resolved
[]

### Check List
- [ ] New functionality includes testing.
  - [x]  All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [ ] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: oneview/config.go [hunk 1]
```diff
@@ -41,9 +41,16 @@ func (c *Config) loadAndValidate() error {
...
-
 	c.ovClient = client
-
+	apiver, _ := c.ovClient.GetAPIVersion()
+	//If no api version is provided use the current version to create client
+	if c.OVAPIVersion == 0 {
+		c.OVAPIVersion = apiver.CurrentVersion
+	} else if c.OVAPIVersion < apiver.MinimumVersion {
+		//Throw error if provided api version is not supported
+		return errors.New(fmt.Sprintf("The minimum api version supported is %d", apiver.MinimumVersion))
```
[the remark this answers is not in the document]
Comment (nabhajit-ray) [reply oneview/config.go]: Fixed

File: oneview/config.go [hunk 2]
```diff
@@ -41,9 +41,21 @@ func (c *Config) loadAndValidate() error {
...
 	client := (&ov.OVClient{}).NewOVClient(c.OVUsername, c.OVPassword, c.OVDomain, c.OVEndpoint, c.OVSSLVerify, c.OVAPIVersion, c.OVIfMatch)
-
 	c.ovClient = client
-
+	apiver, err := c.ovClient.GetAPIVersion()
+
+	//If no api version is provided use the current version to create client
+	if c.OVAPIVersion == 0 {
+		if err != nil {
+			return errors.New(fmt.Sprintf("Could not fetch the appliance %s api version", c.OVEndpoint))
```
[the remark this answers is not in the document]
Comment (nabhajit-ray) [reply oneview/config.go]: Fixed

File: oneview/provider.go [hunk 3]
```diff
@@ -53,7 +53,7 @@ func Provider() terraform.ResourceProvider {
 			"ov_apiversion": {
 				Type:        schema.TypeInt,
 				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("ONEVIEW_OV_API_VERSION", 200),
+				DefaultFunc: schema.EnvDefaultFunc("ONEVIEW_OV_API_VERSION", 0),
```
Comment (VenkateshRavula) [inline oneview/provider.go:56]: what does this mean..?

Comment (nabhajit-ray) [reply oneview/provider.go:56]: Default api version in the schema if the value is not found.  Keeping it 0, we are controlling what to for set the  api version

**Review by VenkateshRavula — APPROVED** [review]

**Review by yuvirani — APPROVED** [review]

**Pull request merged**